datasources/src/main/java/org/opensearch/sql/datasources/service/DataSourceServiceImpl.java

```diff
@@ -134,6 +143,35 @@ private void validateDataSourceMetaData(DataSourceMetadata metadata) {
             + " Properties are required parameters.");
   }
 
+  /**
+   * Replaces the fields in the map of the given metadata.
+   *
+   * @param dataSourceData
+   * @param metadata {@link DataSourceMetadata}.
+   */
+  private void replaceOldDatasourceMetadata(
+      Map<String, Object> dataSourceData, DataSourceMetadata metadata) {
+
+    for (String key : dataSourceData.keySet()) {
+      switch (key) {
+          // Name and connector should not be modified
+        case DESCRIPTION_FIELD:
+          metadata.setDescription((String) dataSourceData.get(DESCRIPTION_FIELD));
+          break;
+        case ALLOWED_ROLES_FIELD:
+          metadata.setAllowedRoles((List<String>) dataSourceData.get(ALLOWED_ROLES_FIELD));
+          break;
+        case PROPERTIES_FIELD:
+          Map<String, String> properties = new HashMap<>(metadata.getProperties());
+          properties.putAll(((Map<String, String>) dataSourceData.get(PROPERTIES_FIELD)));
+          break;
+        case NAME_FIELD:
```

Comment:
Throw Exception Here?

Reply:
This will always have `NAME_FIELD`, and `CONNECTOR_FIELD`, since it is pulling this information from the Data Source record. We should just ignore it here/no-op, wdyt?

Reply:
Name is enough right, are we doing any extra validation if connector_field passed is changed?

Reply:
Added some documentation around this - I think it's enough to just skip it. [87f0605](https://github.com/opensearch-project/sql/pull/2273/commits/87f0605414480b577d297b3966e92c7c9df27248)